torch/onnx/utils.py

```diff
@@ -661,6 +661,10 @@ def _export(model, args, f, export_params=True, verbose=False, training=None,
             fixed_batch_size=False, custom_opsets=None, add_node_names=True,
             onnx_shape_inference=True, export_modules_as_functions=False):
 
+    if export_modules_as_functions is not False and opset_version < 15:
```

Comment:
```suggestion
    if export_modules_as_functions and opset_version < 15:
```

Reply:
Any reason we shouldn't use this more pythonic way?

Reply:
to explicitly match the default value `False`, separating from empty set/list.

Reply:
How does the behavior differ between empty set and False?

(And if we really do want to check for False, use `==` rather than `is`)

Reply:
not different for the current version. I recall for a previous version empty list once means export all modules as functions. I'm thinking to set this check as forbidding the use of this argument completely, rather than raising only when detecting local function in graph, for opset version < 15.

Reply:
why is `==` better than `is` in this occasion?

Reply:
I think any value that evaluates to False should be treated the same.
So just `if export_modules_as_functions and opset_version < 15:`

This is the most pythonic and intuitive I think.
Unless there's a good reason for anything more complicated?

As to why not to use `is`: basically using `is` for comparison to anything other than None is unnecessarily and surprisingly specific. Two objects that behave the same in all respects (including `==`) should be handled the same by most code unless there's a good reason.

Here's an example where using `is` may result in surprising behavior. The caller might pass something in that is `== False` and then be surprised.

```python
def foo(x):
  if x is False:
    print("x is False")
  if x == False:
    print("x equals False")

foo(0)
```

prints: `x equals False`

Reply:
Okay I agree we can relax to treat the same for all values that evaluates to False in this case.

I was actually trying to avoid `== False` in this specific case since it would return the same thing for objects with different types. If empty list/set bear different meanings than bool value False (which they did), it would slip through this check. But then again it was a better choice to remove those behaviors.